open_rarity/models/token.py

```diff
@@ -25,5 +25,42 @@ class Token:
     token_standard: TokenStandard
     metadata: TokenMetadata
 
+    def __init__(
```

Comment:
Why we are not using https://docs.python.org/3/library/dataclasses.html#post-init-processing here?